activestorage/test/test_helper.rb

```diff
@@ -14,15 +13,97 @@
 
 require "active_record/testing/query_assertions"
 
-require "active_job"
+require "rails"
+require "active_record/railtie"
+require "active_storage/engine"
+
+ActiveStorage::FixtureSet.file_fixture_path = File.expand_path("fixtures/files", __dir__)
+ActiveStorage.verifier = ActiveSupport::MessageVerifier.new("Testing")
+
+module ActiveStorage
+  class TestApp < Rails::Application
+    config.eager_load = false
+    config.root = File.join(__dir__, "support")
+    config.fixture_paths = [File.expand_path("fixtures", __dir__)]
+    config.autoload_paths << File.join(__dir__, "support", "models")
+
+    config.active_storage.service = :local
+    # Variant tracking has been true since load_defaults(6.1)
+    # However, several tests depend on the defaults being loaded from the old dummy app config.
+    # Since this is the only config option that was dependent upon that assumption, we can keep it here.
+    config.active_storage.track_variants = true
+
+    # FIXME: need to disable CSRF protection for this test in particular:
+    # test/controllers/direct_uploads_controller_test.rb:128 ("creating new direct upload")
+    config.action_controller.allow_forgery_protection = false
+  end
+end
+
+service_configs = begin
+  ActiveSupport::ConfigurationFile.parse(File.expand_path("service/configurations.yml", __dir__)).deep_symbolize_keys
```

Comment:
Note, it seems like we currently aren't running the other service configurations in CI:
https://buildkite.com/rails/rails/builds/103922#018d0be6-9c18-4624-bba6-a43c4d36bf23/1205-1214

I was hoping to test this path with this refactor.

Reply:
Ok, this hasn't been working since #47259:
https://buildkite.com/rails/rails/builds/93948#018666ba-a8de-47d3-877d-26788ab11fcd/1010-1083

Reply:
Fixed in #50787